src/mongocxx/options/server_api.hpp

```diff
@@ -0,0 +1,142 @@
+// Copyright 2021 MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <string>
+
+#include <bsoncxx/stdx/optional.hpp>
+#include <bsoncxx/stdx/string_view.hpp>
+#include <mongocxx/stdx.hpp>
+
+#include <mongocxx/config/prelude.hpp>
+
+namespace mongocxx {
+MONGOCXX_INLINE_NAMESPACE_BEGIN
+
+class client;
+class pool;
+
+namespace options {
+
+///
+/// Class representing options for server API.
+///
+class MONGOCXX_API server_api {
+   public:
+    ///
+    /// Enum representing the possible values for server API version.
+    ///
+    enum class version { k_version_1 };
+
+    ///
+    /// Converts a version enum value to its string value.
+    ///
+    /// @param version
+    ///   The enum value to convert to a string.
+    ///
+    /// @return
+    ///   The string value of the given enum value.
+    ///
+    static std::string version_to_string(version version);
+
+    ///
+    /// Converts a version string to its enum value.
+    ///
+    /// @param version
+    ///   The string to convert to an enum value.
+    ///
+    /// @return
+    ///   The enum value of the given string.
+    ///
+    static version version_from_string(stdx::string_view version);
+
+    ///
+    /// Constructs a new server_api object.
+    ///
+    /// The specified API version will be sent to the server. This will cause
+    /// the server to behave in a manner compatible with that API version.
+    /// It also causes the driver to behave in a manner compatible with the
```

Comment:
> driver to behave in a manner compatible with the driver's behavior as of the release when the driver

Reads a little strange to me. It looks like this is close to the [Background](https://github.com/mongodb/specifications/blob/master/source/versioned-api/versioned-api.rst#background), perhaps it could use that directly:

> The driver will behave in a manner compatible with a server configured with that API version, regardless of the server's actual release version.

Reply:
Hmm good point; used that documentation instead.